clang/lib/CIR/Lowering/ThroughMLIR/LowerCIRToMLIR.cpp

```diff
@@ -153,6 +166,11 @@ class CIRConstantOpLowering
     if (mlir::isa<mlir::cir::BoolType>(op.getType())) {
       auto boolValue = mlir::cast<mlir::cir::BoolAttr>(op.getValue());
       value = rewriter.getIntegerAttr(ty, boolValue.getValue());
+    } else if (op.getType().isa<mlir::cir::CIRFPTypeInterface>()) {
+      assert(ty.isF32() || ty.isF64() && "NYI");
```

Comment:
I missed this before I merged. During the lowering passes, you should return `mlir::failure()` for these instead, since it provides a more clean error message. If you could address that in another PR it'd be great.

Reply:
OK, @bcardosolopes I will do that in next PR